inspirehep/modules/tools/utils.py

```diff
@@ -22,134 +22,30 @@
 
 """Utility functions for various tools."""
 
-from __future__ import absolute_import, division, print_function
-
-import re
-
-import six
-
-from inspire_schemas.api import LiteratureBuilder
-
-from refextract.documents.pdf import replace_undesirable_characters
-from refextract.documents.text import wash_line
-
-# Differentiate authors from affiliations:
-# authors have a number after name, affiliations lines start with a number
-split_authors_affs_pattern = re.compile(r'(.*?\d)\n+(\d.*)', flags=re.DOTALL)
-# Differentiate affiliations from one another, line break is required
-split_affs_pattern = re.compile(r'(\d+\.?[\s\t]*.*)\n*', flags=re.UNICODE)
-
 
 def authorlist(text):
-    """Call the relevant formatter depending on the existence of affiliations.
+    """
+    Return a record containing the authors parsed from text.
+    """
 
-    Input can be in two different formats:
-        F. Lastname, F.M. Otherlastname
-    or
-        F. Lastname1, F.M. Otherlastname1,2
-        1 CERN
-        2 Otheraffiliation
+    import six
+    from inspirehep.modules.tools.authortool import create_authors
+    from inspire_schemas.api import LiteratureBuilder
+    from refextract.documents.pdf import replace_undesirable_characters
 
-    There should always be only one affiliation per line and the affiliation
-    ids in author names should be separated with commas.
-    """
-    if not text:
-        return {}
+    builder = LiteratureBuilder()
+
+    # Do some pre-cleaning of the input string
     if not isinstance(text, six.text_type):
         text = text.decode('utf-8')
-    # Do some pre-cleaning of the input string
-    text = text.replace('\r', '')  # Input from the form contains unwanted \r's
-    text = text.replace(u'†', '')
-    text = text.replace(u'∗', '')
-    text = re.sub(r'( *\n+ *)', r'\n', text)
     text = replace_undesirable_characters(text)
-    text = wash_line(text)
-
-    # Assume that if there are numbers in author name, they are affilation ids.
-    # Also assume that if one author has an affiliation, everyone else has too.
-    if re.search(r'([a-zA-Z_-]+[\n]*\d+)', text):
-        return authorlist_with_affiliations(text)
-    return authorlist_without_affiliations(text)
-
+    text = text.replace('\r', '')  # Input from the form contains unwanted \r's
 
-def authorlist_without_affiliations(text):
-    """Return a record containing the authors."""
-    builder = LiteratureBuilder()
-    authors = text.replace(' and ', ', ').split(', ')
-    for author in authors:
-        builder.add_author(builder.make_author(author))
+    for fullname, author_affs in create_authors(text):
+        if author_affs:
```

Comment:
The `if` is not needed, the builder can deal with empty arguments.

Reply:
great - thanks